cranelift/entity/src/list.rs

```diff
@@ -639,6 +640,19 @@ impl<T: EntityRef + ReservedValue> EntityList<T> {
     }
 }
 
+#[cfg(feature = "bytemuck")]
+mod bytemuck_impl {
+    use crate::packed_option::ReservedValue;
+    use crate::{EntityList, EntityRef};
+    use bytemuck::{Pod, Zeroable};
+
+    // Soundness: `EntityList` is a `repr(C)` struct
+    // containing only a `u32` and a zero-sized `PhantomData`,
+    // so it can always implement `Pod` regardless of `T`.
```

Comment:
This is a valid argument for the soundness of `Pod`, but not of `Zeroable`, as far as I can tell, right? In fact an all-zero-bits `EntityList` is not guaranteed to be valid at all, and we don't otherwise allow the existence of `EntityList` values that are "dangling"...

Reply:
Ah, yes, you're right; I had forgotten that we use zero as an empty-list sentinel here. Thanks!